### PR TITLE
adds FrontC.3.4.3

### PR DESCRIPTION
**Pull request by ivg**
This is a small update that adds support for pure-asm bodies and provies better diagnostic messages in case of parsing errors. It also cleans a bit the installation procedure (moves META to the upstream repository and applies the opam.patch). It also builds and installs the cmxs file (and updates, correpsondingly, the META file). 

Changelog:

* BinaryAnalysisPlatform/FrontC#7 raises parse errors rather than asserting false
* BinaryAnalysisPlatform/FrontC#11 supports functions with pure-asm bodies
* BinaryAnalysisPlatform/FrontC#13 builds and installs cmxs

### CODE REVIEW

File: packages/FrontC/FrontC.3.4.3/opam [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+homepage: "https://github.com/BinaryAnalysisPlatform/FrontC"
+bug-reports: "https://github.com/BinaryAnalysisPlatform/FrontC/issues"
+dev-repo: "git+https://github.com/BinaryAnalysisPlatform/FrontC"
+maintainer: "Ivan Gotovchits <ivg@ieee.org>"
+license: "LGPL-2.1-only"
+depends: [
+  "ocaml"
+]
+install: [
+  [make "install" "PREFIX=%{lib}%" "OCAML_SITE=%{lib}%"]
```
Comment (kit-ty-kate) [inline packages/FrontC/FrontC.3.4.3/opam:12]: I'm not seeing any `build` section in this package. Are there nothing built? If so, please use it, putting everything in the `install` section will slow down concurrent installation for your users'

Comment (ivg) [reply packages/FrontC/FrontC.3.4.3/opam:12]: good point! ALso, is there any sense in `FrontC.install` file? I have inherited it from the previous maintainers of FrontC and have no idea what it is meant for. Any guesses?

**Comment by ivg** [comment]
@kit-ty-kate, the reporter has acted :)

**Comment by kit-ty-kate** [comment]
Sorry I missed it. Thanks!

**Pull request merged**